Code/ForceField/ANI/AtomicContrib.cpp

```diff
@@ -132,7 +133,39 @@ double ANIAtomContrib::getEnergy(Eigen::ArrayXXd &aev) const {
   return this->ANIAtomContrib::forwardProp(row) + this->d_selfEnergy;
 }
 
-void ANIAtomContrib::getGrad(double *pos, double *grad) const {}
+void ANIAtomContrib::getGrad(double *pos, double *grad) const {
+  auto initEnergy = this->dp_forceField->calcEnergy(pos);
```

Comment:
You are calculating this energy but not using it.
You could switch to a three-point approximation to the gradient (which might be more accurate?) but if not, you can stop doing this calculation.